### DISCUSSION TITLE
Implementing AUTH_MODEL for Import & Export Enhancement

### DISCUSSION THREAD

**Discussion by SupianIDz** [filamentphp/filament#11958] [Ideas]
Today I was confused when using the database on `QUEUE_CONNECTION` Import & Export Job always failed, after I investigated because `\App\Models\User` was not found.

FYI: I'm using `\App\Domains\User\Models\User`

Should we use [AUTH_MODEL](https://github.com/laravel/laravel/blob/441845d88dfc2a4fbf59f1e57d48889c8de664bb/config/auth.php#L65) instead of forcing bind `User` model to `Illuminate\Contracts\Auth\Authenticatable` on `ServiceProvider` in Import & Export ?

Or add this in the Documentation?

see : https://github.com/filamentphp/filament/blob/68e4a11153d740b09d1162bbd9182e77499cc736/packages/actions/src/Exports/Models/Export.php#L52-L56

https://github.com/filamentphp/filament/blob/68e4a11153d740b09d1162bbd9182e77499cc736/packages/actions/src/Imports/Models/Import.php#L58-L62

**Comment by Ercogx** [comment]
I also do not understand why binding is needed and not to take the value from the config, I have not seen such use in other libraries

**Comment by ProgrammerZ** [comment]
I found it strange to bind to Authenticatable. I believe a configuration setting would be a better approach, and even better if it defaults to pulling Authenticatable Model from authGuard() config and also custom config if different from default authGuard().

Also consider issue with `app(Router::class)->middlewareGroup('filament.actions', ['web', 'auth']);` when using non standard guard like`['auth:admin']`

**Comment by jazzy1986** [comment]
encounter error on this too, because i use custom SUser.php instead of using default User.php

**Comment by MladenJanjetovic** [comment]
same here, My jobs are breaking because I use Admin model as Filament default guard. And it's all set well in fillement config file but it still uses User model for auth

**Reply by SupianIDz** [reply]
```php
$this->app->bind(\Illuminate\Contracts\Auth\Authenticatable::class, YourUserModel::class)
```

In the `register` method of the Service Provider

**Reply by MladenJanjetovic** [reply]
But will that affect my default web auth guard which is not related to Admin model?
